### DISCUSSION TITLE
How do I associate web3Modal with a provider?

### DISCUSSION THREAD

**Discussion by Jaycelv** [WalletConnect/walletconnect-monorepo#3229] [Developer Support] [answered]
When I was using version 1.8, I could pass WalletConnectProvider as a parameter into Web3Modal, then I could call the connect method of web3Modal to get the provider instance, and then I could listen to the instance. Now how do I associate it? I see the documents are written separately

**Comment by glitch-txs** [comment, marked as answer]
Web3Modal works with Wagmi internally, which at the same time uses [Viem](https://viem.sh/) to interact with the blockchain. Wagmi also has an [ethers.js addapter](https://viem.sh/). We'll add wider support in Web3Modal v3
